### PR TITLE
Don't use HTTP request if ids list is empty

### PR DESCRIPTION
**Pull request by H--o-l**
The goal is to prevent un-necessary HTTP request to the database when the list
of ids is empty.
It's especially useful when use in combination with `Database.update_docs()`:
```py
with db.update_docs([]) as bulk:
    bulk._docs.append(a_document)
    bulk._docs.append(another_document)
```

### CODE REVIEW

File: aiocouch/database.py [hunk 1]
```diff
@@ -64,6 +64,8 @@ def __init__(self, couchdb, id):
         await self._delete()
 
     async def docs(self, ids=None, create=False, prefix=None, **params):
+        if ids is not None and not ids:
```
Comment (adrienverge) [inline aiocouch/database.py:67]: May I suggest:
```suggestion
        if ids is not None and len(ids) == 0:
```
? 

I think it's a bit clearer for developers.

File: aiocouch/database.py [hunk 2]
```diff
@@ -64,6 +64,8 @@ def __init__(self, couchdb, id):
...
+            return
```
Comment (bmario) [inline aiocouch/database.py:68]: I'm inclined to say that passing an empty list for ids should raise a ValueError instead.

File: aiocouch/database.py (hunk 1 above)
Comment (H--o-l) [reply aiocouch/database.py:67]: Thanks for the feedback @adrienverge, I updated the commit accordingly.

**Comment by H--o-l** [comment]
> I'd also appreciate a new test case.

I understand, and I tried to do it: I tried to mock `Database.View._get()` in order to ensure it is not call with the new code.
But I didn't manage to pull it off in a reasonable amount of time.
But maybe my idea wasn't the right one.

(PS: If finally we choose to raise a `ValueError`, in that case I will able to update tests accordingly myself.)

File: aiocouch/database.py (hunk 2 above)
Comment (H--o-l) [reply aiocouch/database.py:68]: Ok, but in that case I will not be able to use the following code:
```py
with db.update_docs([]) as bulk:
    bulk._docs.append(a_document)
    bulk._docs.append(another_document)
```
It's sad for me, because this code is useful in the situation where I know a bunch of documents ids, I have them, and I want to save them, all at once.

**Comment by bmario** [comment]
> I understand, and I tried to do it: I tried to mock `Database.View._get()` in order to ensure it is not call with the new code.
> But I didn't manage to pull it off in a reasonable amount of time.
> But maybe my idea wasn't the right one.

For now, I'd be fine with just a test calling the code path and checking that it indeed raises the ValueError. I can imagine that adding a mock would be a quite daunting task.

> (PS: If finally we choose to raise a `ValueError`, in that case I will able to update tests accordingly myself.)

If you can't come up with a good example, why that is a bad idea, I'd say so.

File: aiocouch/database.py (hunk 2 above)
Comment (bmario) [reply aiocouch/database.py:68]: That's what `db.create_docs()` was introduced for recently.

**Comment by H--o-l** [comment]
Hi @bmario !

Sorry I slowly but surely release that I didn't fully understand https://github.com/metricq/aiocouch/pull/26#issue-397009862

What I mean is that the example that made me open this PR, which was the following:
```py
with db.update_docs([]) as bulk:  # do a HTTP call that I want to avoid
    bulk._docs.append(a_document)
    bulk._docs.append(another_document)
```
Could have simply be 
```py
with db.create_docs() as bulk:  # no HTTP, perfect!
    bulk._docs.append(a_document)
    bulk._docs.append(another_document)
```

On the side of raising a `ValueError` when `update_docs()` is called with `ids=[]`, I think that `[]` is a value like any other, and should not raise an error.
More generally, if a function applied to a list, and raise an error in the special case were that list is empty, it makes it hard to use it combined in the middle of other transformation of that list.
For example, in functional programming you can do things like that:
```
new_list = list.find(...).filter(...).merge(another_list).map(...)...
```
If one of these functions raise an error on the empty list special case, it complexity writing that kind of code a lot.
I'm not sure I'm understandable, don't hesitate to tell me if you don't understand or disagree.

In the meantime I'm for closing this PR because the use case (prevent un-necessary HTTP call) can be made with `create_docs()` function.

Thanks for your feedback!

**Comment by bmario** [comment]
I still think that this PR has value. As you correctly brought to my attention that the following code triggers an unnecessary request:
```python
with db.update_docs([]) as bulk:
   pass
```

> I think that `[]` is a value like any other, and should not raise an error.

This argument does not convince me. The code expresses the wish to retrieve existing documents, alter them, and store the updated contents. But the set of documents is empty. That doesn't sound right.

> More generally, if a function applied to a list, and raise an error in the special case were that list is empty, it makes it hard to use it combined in the middle of other transformation of that list.

Yes, this thinking led me to the alternative suggestion of ~~returning an empty list instead of the implicit `None`, because, under normal conditions, the function is a generator.~~ Never mind, TIL `return` in a generator does the right thing™.

So in short, just add a test, that `len(db.docs([])) == 0` and I'm willing to merge this.

**Comment by H--o-l** [comment]
Hi @bmario !
Ok, that sounds good.

There is already a test for `db.docs([])` [here](https://github.com/metricq/aiocouch/blob/9d9ae99ba6b3b44b11358c59822529d3982de0d6/tests/test_database.py#L145-L148), thus I added one for `db.update_docs([])`.
I let you see, don't hesitate to criticize.
